src/Compilers/CSharp/Test/Symbol/Symbols/Source/EventTests.cs

```diff
@@ -325,6 +325,397 @@ class C
+        public void EventWithoutDynamicAttributeFromLegacyCompiler()
+        {
+            // The important thing here is that the .event doesn't have a DynamicAttribute, but the add and remove accessors do.
+            // This is how the native compiler (4.6) emits "event Action<dynamic>".
+            var ilSource = @"
+.assembly extern System.Core
+{
+  .publickeytoken = (B7 7A 5C 56 19 34 E0 89 )                         // .z\V.4..
+  .ver 4:0:0:0
+}
+
+.class public auto ansi beforefieldinit C
+       extends [mscorlib]System.Object
+{
+  .field private class [mscorlib]System.Action`1<object> E
+  .custom instance void [System.Core]System.Runtime.CompilerServices.DynamicAttribute::.ctor(bool[]) = ( 01 00 02 00 00 00 00 01 00 00 ) 
+  .method public hidebysig specialname instance void 
+          add_E(class [mscorlib]System.Action`1<object> 'value') cil managed
+  {
+    .param [1]
+    .custom instance void [System.Core]System.Runtime.CompilerServices.DynamicAttribute::.ctor(bool[]) = ( 01 00 02 00 00 00 00 01 00 00 ) 
+    // Code size       2 (0x2)
+    .maxstack  8
+    IL_0000:  nop
+    IL_0001:  ret
+
+  } // end of method C::add_E
+
+  .method public hidebysig specialname instance void 
+          remove_E(class [mscorlib]System.Action`1<object> 'value') cil managed
+  {
+    .param [1]
+    .custom instance void [System.Core]System.Runtime.CompilerServices.DynamicAttribute::.ctor(bool[]) = ( 01 00 02 00 00 00 00 01 00 00 ) 
+    // Code size       2 (0x2)
+    .maxstack  8
+    IL_0000:  nop
+    IL_0001:  ret
+  } // end of method C::remove_E
+
+  .method public hidebysig specialname rtspecialname 
+          instance void  .ctor() cil managed
+  {
+    // Code size       7 (0x7)
+    .maxstack  8
+    IL_0000:  ldarg.0
+    IL_0001:  call       instance void [mscorlib]System.Object::.ctor()
+    IL_0006:  ret
+  } // end of method C::.ctor
+
+  .event class [mscorlib]System.Action`1<object> E
+  {
+    .addon instance void C::add_E(class [mscorlib]System.Action`1<object>)
+    .removeon instance void C::remove_E(class [mscorlib]System.Action`1<object>)
+  } // end of event C::E
+} // end of class C
+";
+            var source = @"
+class Module1
+{
+    static void Main()
+    {
+        C x = new C();
+        x.E += (f) => f.Bogus(); // Type of E is seen as Action<object>, not Action<dynamic>
+    }
+}
+";
+            var compilation = CreateCompilationWithCustomILSource(source, ilSource, new[] { CSharpRef, SystemCoreRef }).VerifyDiagnostics(
+                // (7,25): error CS1061: 'object' does not contain a definition for 'Bogus' and no extension method 'Bogus' accepting a first argument of type 'object' could be found (are you missing a using directive or an assembly reference?)
+                //         x.E += (f) => f.Bogus();
+                Diagnostic(ErrorCode.ERR_NoSuchMemberOrExtension, "Bogus").WithArguments("object", "Bogus").WithLocation(7, 25));
```

Comment:
I think this test is actually supposed to test diagnostics, we want to make sure we are reporting the same/similar error for this scenario as before. Try replacing the lambda with ```null```.